vortex-array/src/data/mod.rs

```diff
@@ -146,6 +146,13 @@ impl ArrayData {
         }
     }
 
+    /// Return whether the array is constant.
+    pub fn is_constant(&self) -> bool {
+        self.statistics()
+            .get_as::<bool>(Stat::IsConstant)
```

Comment:
this should `compute_as` instead of `get_as`, otherwise too subtle a footman

Reply:
We basically never want compute_as.

I can rename it definitely_is_constant? But kept is consistent with as_constant.

Reply:
Generally true, but not sure I agree re: `Stat::IsConstant` specifically given how much we optimize for it and that it short-circuits

Reply:
Hmmm. Definitely worth an experiment. If we can separate it out and implement it with a short-circuit for all encodings, then we can flip this to be compute_as and see what happens.

We can also cache stats for ArrayView if we really want.

Reply:
I audited the stats implementations and split them up last week, only (admittedly big) exception right now is varbin & varbinview

Reply:
https://github.com/spiraldb/vortex/issues/1411